extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSourceFactory.java

```diff
@@ -77,8 +75,8 @@ public DataSource createSource(DataFlowStartMessage request) {
 
         return S3DataSource.Builder.newInstance()
                 .bucketName(source.getStringProperty(BUCKET_NAME))
-                .keyName(source.getKeyName())
-                .keyPrefix(source.getStringProperty(KEY_PREFIX))
+                .objectName(source.getStringProperty(OBJECT_NAME))
+                .objectPrefix(source.getStringProperty(OBJECT_PREFIX))
```

Comment:
this looks like a breaking change, because assets data address that are currently stored will stop working with the new version (because they don't have the OBJECT_NAME property. 
Please make it work also with the old structure and eventually log a warning so there will be time to adapt,
otherwise a data migration would be required but it could be harder to apply than a deprecation.

Reply:
> this looks like a breaking change, because assets data address that are currently stored will stop working with the new version (because they don't have the OBJECT_NAME property. Please make it work also with the old structure and eventually log a warning so there will be time to adapt, otherwise a data migration would be required but it could be harder to apply than a deprecation.

I deprecated the properties and added logic to make code backwards compatible.